tuf/api/metadata.py

```diff
@@ -5,7 +5,21 @@
 
 This module provides container classes for TUF role metadata, including methods
 to read and write from and to file, perform TUF-compliant metadata updates, and
-create and verify signatures.
+create and verify signatures. The intent of the API is to provide safe
+serialization and deserializaiton and a safe API for accessing and modifying
+the metadata.
+
+Each of the top level metadata roles is represented by a separate class and each
+of those classes denotes exactly one metadata file.
```

Comment:
This feels a bit vague (for someone seeing Metadata API for the first time) and inaccurate: Root does not map to a root metadata file, it maps to the signed portion of the metadata file... Metadata is the equivalent of a metadata file.

I'd like to instead see actual names of classes (with links directly to the relevant doc), explanation of Root,Targets objects vs Metadata object and how to work with them. Maybe you could do it with an example of how `Metadata[Root]` represents the TUF root role and that in practice it means a Metadata object with a signed of type Root. This is the difficult and weird part of Metadata API and requires an explanation I think.